### DISCUSSION TITLE
How important is the fragment file? What does it add to downstream analysis?

### DISCUSSION THREAD

**Discussion by HaglundA** [stuart-lab/signac#624] [Q&A] [answered]
Thanks for developing this tool. I had a quick question regarding the fragment file.

My understanding of what the fragment file is:
- contains all fragments from the scATAC-seq data generation 
- As a result this is a very large file
- The feature matrix is generated from the fragment file by your mapping pipeline (let's say from 10X genomics)

My issue at the moment is that when I include the fragment file in "CreateChromatinAssay", as opposed to the feature matrix alone, the command ends up taking 8+ hours - whereas if I exclude it the Assay gets generated in minutes. 

What I don't understand completely is what information I am missing here? Is it a way to correct for the peaks observed? Or rather to add complexity genomewide in regions where peaks might not appear?

Any comments would be greatly appreciated!

**Comment by timoast** [comment, marked as answer]
It's very unusual for it to take so much additional time when including the fragment file. Can you post the full code that you're running, and the number of cells in your experiment?

The fragment file is used in several steps, including:
* Computing QC metrics (nucleosome signal, TSS enrichment, fraction counts in blacklist regions)
* Quantifying counts in genomic regions (constructing new matrices)
* Plotting coverage tracks

**Reply by HaglundA** [reply]
Thanks for the very quick reply! 

I've been working via Jupyter notebook and running the cell in which I create the chromatin assay leaves it running forever - when running it on the command line I realise this was a library issue (for which I get the error straight away). Case solved, and apologies for wasting your time! 

Thank you again for your quick and detailed answer.

**Reply by timoast** [reply]
Ah yes, this can be an issue when running jupyter notebooks...errors sometimes seem to get buried. Glad you could sort it out
